### PR TITLE
docs: new view features

### PR DESCRIPTION
**Pull request by pvlugter**
Refs https://github.com/lightbend/kalix-proxy/issues/1335.

Add simple docs for some of the new features for views. Only targeted at `kalix-main` as these will be in the Kalix proxy.

### CODE REVIEW

File: docs/src/modules/java/pages/views.adoc [hunk 1]
```diff
@@ -452,15 +456,116 @@ Use the following filter predicates to further refine results:
 * `<` less than
 * `\<=` less than or equals
 
-Combine filter conditions with the `AND` and `OR` operators.
+==== Logical operators
+
+Combine filter conditions with the `AND` and `OR` operators, and negate using the `NOT` operator. Group conditions using parentheses. Note that `AND` has precedence over `OR`.
```
Comment (patriknw) [inline docs/src/modules/java/pages/views.adoc:461]: don't we reject OR until that index issue has been fixed?

File: docs/src/modules/java/pages/views.adoc [hunk 2]
```diff
@@ -452,15 +456,116 @@ Use the following filter predicates to further refine results:
...
 
 [source,proto,indent=0]
 ----
 SELECT * FROM customers WHERE
   name = :customer_name AND address.city = 'New York' OR
-  name = :customer_name AND address.city = 'San Francisco'
+  NOT (name = :customer_name AND address.city = 'San Francisco')
+----
+
+NOTE: `OR` support is currently disabled, until it can be more efficiently indexed.
```
Comment (patriknw) [inline docs/src/modules/java/pages/views.adoc:470]: maybe move this to the top of the section?

**Review by johanandren — APPROVED** [review]

File: docs/src/modules/java/pages/views.adoc [hunk 3]
```diff
@@ -452,15 +456,116 @@ Use the following filter predicates to further refine results:
...
+
+==== Pattern matching
+
+Use `LIKE` to pattern match on strings. The standard SQL `LIKE` patterns are supported, with `_` (underscore) matching a single character, and `%` (percent sign) matching any sequence of zero or more characters.
+
+----
+SELECT * FROM customers WHERE name LIKE 'Bob%'
 ----
 
+NOTE: For index efficiency, only constant patterns with literal strings are supported, and the pattern must have a non-wildcard prefix or suffix. A pattern like `'%foo%'` is not supported.
```
Comment (patriknw) [inline docs/src/modules/java/pages/views.adoc:502]: Do we need to clarify that request parameters can't be included in the pattern? You say that indirectly by "only constant patterns with literal strings" but maybe another addition at the end could make that even more clear?

File: docs/src/modules/java/pages/views.adoc [hunk 4]
```diff
@@ -498,4 +603,12 @@ To check if there are more pages left, you can use the function `has_more()` pro
...
\ No newline at end of file
+This query will return `more_customers = true` when the view contains more than 10 customers.
+
+==== Total count of results
+
+To get the total number of results that will be returned over all pages, use `COUNT(*)` in a query that projects its results into a field. The total count will be returned in the aliased field (using `AS`) or otherwise into a field named `count`.
+
+----
+SELECT * AS customers, COUNT(*) AS total, has_more() AS more FROM customers LIMIT 10
+----
```
Comment (patriknw) [inline docs/src/modules/java/pages/views.adoc:614]: do we need another example of COUNT(*) where it's not used together with paging?

**Review by patriknw — COMMENTED** [review]
looking good

File: docs/src/modules/java/pages/views.adoc [hunk 5]
```diff
@@ -452,15 +456,116 @@ Use the following filter predicates to further refine results:
...
+
+If the query contains multiple words, the text search will find values that contain all of these words (logically combined with AND), with tokenization and normalization automatically applied.
+
+The following text search language configurations are supported: `'danish'`, `'dutch'`, `'english'`, `'finnish'`, `'french'`, `'german'`, `'hungarian'`, `'italian'`, `'norwegian'`, `'portuguese'`, `'romanian'`, `'russian'`, `'simple'`, `'spanish'`, `'swedish'`, `'turkish'`. By default, a `'simple'` configuration will be used, without language-specific features.
+
+----
+SELECT * FROM customers WHERE text_search(profile, :search_words, 'english')
+----
+
```
[the remark this answers is not in the document]
Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc:519]: Good idea, will add something

File: docs/src/modules/java/pages/views.adoc (hunk 2 above)
Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc:470]: Yeah, this one is awkward at the moment. Might be good to comment out the full text, and have something that doesn't demonstrate OR as well. I'll update.

File: docs/src/modules/java/pages/views.adoc (hunk 4 above)
Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc:614]: This special `COUNT(*)` is mainly for paging. It currently requires the `* AS results` projection into repeated field. Not much use for the total count outside of a query with a LIMIT.

I did mention in the original issue whether we support aggregating functions more generally.

File: docs/src/modules/java/pages/views.adoc (hunk 3 above)
Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc:502]: Ok, will massage.

File: docs/src/modules/java/pages/views.adoc (hunk 4 above)
Comment (patriknw) [reply docs/src/modules/java/pages/views.adoc:614]: ok, so we don't support something like this?
```
SELECT COUNT(*) FROM customers
```

Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc:614]: Right, that's not currently supported.

File: docs/src/modules/java/pages/views.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc:528]: added note

File: docs/src/modules/java/pages/views.adoc (hunk 1 above)
Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc:462]: removed OR

File: docs/src/modules/java/pages/views.adoc (hunk 2 above)
Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc:474]: Removed OR. Commented out the original.

File: docs/src/modules/java/pages/views.adoc (hunk 3 above)
Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc]: Added explicitly that request parameters can't be used.

File: docs/src/modules/java/pages/views.adoc (hunk 4 above)
Comment (pvlugter) [reply docs/src/modules/java/pages/views.adoc:625]: Created a separate issue for possibly extending this: https://github.com/lightbend/kalix-proxy/issues/1348

**Review by johanandren — APPROVED** [review]

**Pull request merged**